RFC/src/RFC-0203_StealthAddresses.md

```diff
@@ -0,0 +1,115 @@
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This Request for Comment (RFC) presents the implementation of Dual-Key Stealth Addresses in One-Sided payments to improve
+privacy for receivers of these payments on the Tari base layer.
+
+## Related Requests for Comment
+
+- [RFC-0201: TariScript](RFC-0201_TariScript.md)
+
+## Introduction
+
+The Tari protocol extends the [Mimblewimble] protocol to include scripting in the form of [TariScript]. One of the 
+first features implemented using [TariScript] was [one-sided payments]. These are payments to a recipient that do not 
+require an interactive negotiation in the same way a standard [Mimblewimble] transaction does. One of the main downsides
+of the current implementation of [one-sided payments] is that the script key used is the Public Key of the recipient's 
+wallet. This public key is embedded in the [TariScript] of the [UTXO] created by the sender. The issue is that it becomes
+very easy for a third party to scan the blockchain to look for one-sided transaction outputs being sent to a given wallet. 
+In order to alleviate this privacy leak, this RFC proposes the use of Dual-Key Stealth Addresses to be used as the script
+key when sending a one-sided payment.
+
+## Brief background on the development of Stealth Addresses
+
+Stealth addresses were first proposed on the Bitcoin Talk forum by user [Bytecoin]. The concept was further refined in 
+the [Cryptonote] whitepaper and by [Peter Todd] which went on to be used in Monero. These formulations were very similar 
+to the [BIP-32] style of address generation. Later in 2014 a developer called rynomster/sdcoin proposed a further 
+improvement to the scheme that he called the Dual-Key Stealth Address Protocol (DKSAP) that allowed for a separate 
+scanning key and spending key. Since then there have been many variations of DKSAP proposed, but generally they only 
+offer performance optimizations for certain scenarios or add an application-specific feature. For our application, DKSAP 
+will do the job.
+
+## Dual-key Stealth Addresses
+
+The Dual-key Stealth Address Protocol (DKSAP) uses two key-pairs for the recipient of a [one-sided transaction], 
+\\( A = a \cdot G \\) and \\( B = b \cdot G \\). Where \\( a \\) is called the scan key and \\( b \\) is the spend key.
+A recipient will distribute the public keys out of band to receive [one-sided payments].
+
+The protocol that a sender will use to make a payment to the recipient is as follows:
+1. Sender generates a random nonce key-pair \\( R = r \cdot G \\).
+2. Sender calculates a ECDH shared secret \\(c = H( r \cdot a \cdot G ) = H( a \cdot R) = H( r \cdot A) \\), where
+\\( H( \cdot ) \\) is a cryptographic hash function.
+3. The sender will then use \\( K_s = c \cdot G + B \\) as the last public key in the [one-sided payment] script. 
+4. The sender will communicate R to the sender by including it in the script so the recipient can parse it but 
```

Comment:
I think this whole sentence could be a bit clearer with something like: 

"The sender includes R in the output script for the receiver, but it is then dropped since it is not required for the script execution."